File: M/1877-minimize-maximum-pair-sum-in-array/main.go
package main

import (
	"fmt"
	"sort"
)

func minPairSum(nums []int) int {
	sort.Ints(nums)
	min := nums[0] + nums[len(nums)-1]
	for i := 1; i < len(nums)/2; i++ {
		if min < nums[i]+nums[len(nums)-1-i] {
			min = nums[i] + nums[len(nums)-1-i]
		}
	}
	return min
}

func main() {
	fmt.Println()
}

File: M/1877-minimize-maximum-pair-sum-in-array/main.rs
impl Solution {
  pub fn min_pair_sum(nums: Vec<i32>) -> i32 {
    let mut nums = nums;
    nums.sort_unstable();
    (0..nums.len() / 2).fold(0, |ans, idx| {
      ans.max(nums[idx] + nums[nums.len() - 1 - idx])
    })
  }
}
